java/connectors/semantickernel-connectors-memory-jdbc/src/main/java/com/microsoft/semantickernel/connectors/memory/jdbc/JDBCConnector.java

```diff
@@ -135,70 +137,84 @@ public Mono<Void> createCollectionAsync(String collectionName) {
                 .then();
     }
 
-    public Mono<Void> updateAsync(
+    protected String resolveKey(String key) {
+        return key != null && !key.isEmpty() ? key : UUID.randomUUID().toString();
+    }
+
+    public Mono<String> upsertAsync(
             String collection,
             String key,
             String metadata,
             String embedding,
             ZonedDateTime timestamp) {
-
+        final String upsertKey = resolveKey(key);
```

Comment:
Is it consistent with .net that key can be null and will be generated for you? If so add Nullable annotation to the `key` parameter so people know it can be null